File: Wayne/WayneCompiler/src/Dfir_representation/AssociatedRightShiftRegister.java
package Dfir_representation;

public class AssociatedRightShiftRegister{
	   private int parentId;
	   private int nodeId;
	   public AssociatedRightShiftRegister(int NodeId,int parentId){
		   this.parentId = parentId;
		   this.nodeId = NodeId;
	   }
	   public int getParentId(){
		   return this.parentId;
	   }
	   public int getNodeId(){
		   return this.nodeId;
	   }
	   public void print(){
		   System.out.println("NodeId: " + this.nodeId + ' ' + "ParentId: " + this.parentId);
	   }
}

File: Jasmine/JasmineCompiler/Src/DFIR_Compiler/DFIR_Compiler/AssociatedRightShiftRegister.cs
using System;

namespace DFIR_Compiler
{
    public class AssociatedRightShiftRegister
    {
        private int nodeId;
        private int parentId;

        public AssociatedRightShiftRegister(int NodeId, int ParentId)
        {
            nodeId = NodeId;
            parentId = ParentId;
        }

        public virtual int GetNodeId()
        {
            return nodeId;
        }
        public virtual int GetParentId()
        {
            return parentId;
        }

        public virtual void Print()
        {
            Console.WriteLine("NodeId: " + nodeId + ' ' + "ParentId: " + parentId);
        }

    }
}
